app/code/Magento/Sales/Test/Mftf/Test/AdminReorderAfterDeleteCustomerGroupTest.xml

```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="AdminReorderAfterDeleteCustomerGroupTest">
+        <annotations>
+            <features value="Sales"/>
+            <stories value="Github issue: #29059 Error at reorder for order with no longer existing customer group id."/>
+            <title value="Unable to create 'Reorder'."/>
+            <description value="Unable to create 'Reorder' after deleting customer group."/>
+            <severity value="CRITICAL"/>
+            <group value="sales"/>
+        </annotations>
+
+        <before>
+            <createData entity="SimpleProduct2" stepKey="createSimpleProductApi"/>
+            <createData entity="Simple_US_Customer" stepKey="createCustomer" />
+            <actionGroup ref="AdminLoginActionGroup" stepKey="loginAsAdmin"/>
+        </before>
+
+        <after>
+            <deleteData createDataKey="createSimpleProductApi" stepKey="deleteSimpleProductApi"/>
+            <deleteData createDataKey="createCustomer" stepKey="deleteCustomer"/>
+            <actionGroup ref="AdminLogoutActionGroup" stepKey="logoutFromAdmin"/>
+        </after>
+
+        <actionGroup ref="AdminCreateCustomerGroupActionGroup" stepKey="createCustomerGroup">
+            <argument name="groupName" value="{{CustomCustomerGroup.code}}"/>
+            <argument name="taxClass" value="{{CustomCustomerGroup.tax_class_name}}"/>
+        </actionGroup>
+
+        <magentoCLI command="indexer:reindex" stepKey="reindex"/>
+        <magentoCLI command="cache:flush" stepKey="flushCache"/>
```

Comment:
Every time we flush the cache, it introduces long delays in the execution of the test since regenerating all caches is very resource-consuming operation for the system. On the other hand, if we invalidate (not flush) only a particular cache, the process might be 10 times faster. 
I would not recommend flushing the entire cache. This operation is required only for some edge-cases. Clearing a particular (or a few of them) cache type is enough for 99% cases. 

Please, refer to the following example for more details 

https://github.com/magento/magento2/blob/3467046b478dabee1ef3b5ff8dbc9e23c53f6358/app/code/Magento/Sales/Test/Mftf/Test/StorefrontCreateOrdersWithMoveJSCodeBottomTest.xml#L45